### PR TITLE
Fix incorrect if conditions in View

### PR DESCRIPTION
**Pull request by come-nc**
Signed-off-by: Côme Chilliet <come.chilliet@nextcloud.com>

### CODE REVIEW

**Comment by CarlSchwan** [comment]
Could you add a bit more info in the commit message about what this fix and why the previous code was incorrect?

**Comment by come-nc** [comment]
> Could you add a bit more info in the commit message about what this fix and why the previous code was incorrect?

Done.

**Review by CarlSchwan — APPROVED** [review]
This probably only make a difference when you can't read a file but still have permission to do other stuff (file drop)

**Comment by artonge** [comment]
Could we take the opportunity to exit early on this condition instead of wrapping 100 lines into the if: https://github.com/nextcloud/server/blob/dcd2a6841e9f1f111f74c9f6006c3821bfb12e3d/lib/private/Files/View.php#L1446=

**Comment by come-nc** [comment]
> Could we take the opportunity to exit early on this condition instead of wrapping 100 lines into the if: https://github.com/nextcloud/server/blob/dcd2a6841e9f1f111f74c9f6006c3821bfb12e3d/lib/private/Files/View.php#L1446=

Done

**Review by artonge — APPROVED** [review]

File: lib/private/Files/View.php [hunk 1]
```diff
@@ -1435,124 +1435,122 @@ public function getDirectoryContent($directory, $mimetype_filter = '') {
 		if (!Filesystem::isValidPath($directory)) {
 			return [];
 		}
+
 		$path = $this->getAbsolutePath($directory);
 		$path = Filesystem::normalizePath($path);
 		$mount = $this->getMount($directory);
-		if (!$mount) {
```
Comment (CarlSchwan) [inline lib/private/Files/View.php:1441]: Was this removed by error?

Comment (come-nc) [reply lib/private/Files/View.php:1441]: No it was spotted by psalm that this is impossible, getMount will return an object or throw.

Comment (icewind1991) [reply lib/private/Files/View.php:1441]: (it used to sometimes return `null` but that was changed with 24)

**Review by icewind1991 — APPROVED** [review]

**Pull request merged**